src/lib.rs

```diff
@@ -225,22 +225,25 @@ impl Executor {
         // already, then be aggressive: wake all idle threads and spawn one more thread.
         while inner.queue.len() > inner.idle_count * 5 && inner.thread_count < EXECUTOR.thread_limit
         {
-            // The new thread starts in idle state.
-            inner.idle_count += 1;
-            inner.thread_count += 1;
-
-            // Notify all existing idle threads because we need to hurry up.
-            self.cvar.notify_all();
-
             // Generate a new thread ID.
             static ID: AtomicUsize = AtomicUsize::new(1);
             let id = ID.fetch_add(1, Ordering::Relaxed);
 
             // Spawn the new thread.
-            thread::Builder::new()
+            if let Err(e) = thread::Builder::new()
                 .name(format!("blocking-{}", id))
                 .spawn(move || self.main_loop())
-                .unwrap();
+            {
+                log::error!("Failed to spawn new blocking thread: {:?}", e);
```

Comment:
The main problem with the code as-is is that if the thread limit is reached, whenever `grow_pool` is called it will try to spawn a new thread and then immediately bump its head again on the error here, which may be bad for performance/log noise.

Would it be better to add a `working_thread_limit` variable to `Inner` that, by default, is the same as `EXECUTOR.thread_limit`, but whenever an `Err` happens is set to the current number of threads?

Reply:
I went ahead and added this, moving `thread_limit` into the `Inner` guard.